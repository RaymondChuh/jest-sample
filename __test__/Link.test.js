import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../src/Link.js';

test('render successfully', () => {
  const tree = renderer.create(<Link href="www.example.com">Example website</Link>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render an anchor when no page is set', () => {
  const tree = renderer.create(<Link></Link>).toJSON();
  expect(tree).toMatchSnapshot();
});


test('change class when it is hovered', () => {
  const component = renderer.create(<Link href="www.example.com">Example website</Link>)
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
