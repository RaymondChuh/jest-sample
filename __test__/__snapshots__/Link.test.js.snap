exports[`test change class when it is hovered 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Example website
</a>
`;

exports[`test change class when it is hovered 2`] = `
<a
  className="hovered"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Example website
</a>
`;

exports[`test change class when it is hovered 3`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Example website
</a>
`;

exports[`test render an anchor when no page is set 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]} />
`;

exports[`test render successfully 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Example website
</a>
`;
